### DISCUSSION TITLE
[QUESTION] How to display video directly as video thumb like image in chat

### DISCUSSION THREAD

**Discussion by ralphvmo** [flyerhq/flutter_chat_ui#171] [Ideas & Requests]

**Comment by demchenkoalex** [comment]
No way atm, to be implemented

**Comment by otto-dev** [comment]
Would look into whether one can do this with `customMessageBuilder` these days. 

> Build a custom message inside predefined bubble.

See https://github.com/flyerhq/flutter_chat_ui/blob/8175caf6b2efa44be9ae0857aab434c823011351/doc/advanced-usage.md#custom-messages



(I'm currently exploring this library for my own development, so I haven't tried anything yet)

**Reply by demchenkoalex** [reply]
https://github.com/flyerhq/flutter_chat_ui/issues/287 someone already tried here and the results might be not good. In the end Video messages should be supported out of the box, just need to finally finish those replies..

**Reply by otto-dev** [reply]
Thanks for your detailed explanation. Once I reach this stage of development, I will likely look into making a PR to allow insert animations while not rebuilding widgets - unless it's already done at that stage.

**Reply by demchenkoalex** [reply]
Nope it is not done, and as I mentioned somewhere there, simple https://api.flutter.dev/flutter/widgets/AnimatedList-class.html will rebuild every visible message on every insert. So I think it is impossible? Unless we remove AnimatedList (CustomScrollView in our case) and do animation on messages, but I don't think other messages will move if we use animations on messages, but I am not so familiar with Flutter animations :)

**Reply by otto-dev** [reply]
I'm sure there is a way (albeit not with stock AnimatedList). Will look into it in two weeks the earliest, and in eight weeks the latest, based on my current planning.

**Reply by aintend-com** [reply]
Yes, that seems off, I thing in that example the static bee.mp4 uri is used so flutter considers it as const so it does not rebuild it when a new message is added but in my case with dynamic video uri every time the list is rebuilt.

**Reply by otto-dev** [reply]
It was a bug in Flutter, I made a PR upstream, and also one for `flutter_chat_ui` that fixes it in the interim.

https://github.com/flutter/flutter/pull/108710

**Comment by otto-dev** [comment]
Made a PR: https://github.com/flyerhq/flutter_chat_ui/pull/296

If `SliverAnimatedList's` `findChildIndexCallback` were working as expected (as it does for `SliverList`), this would not have been such a pain.

**Reply by otto-dev** [reply]
> If SliverAnimatedList's findChildIndexCallback were working as expected (as it does for SliverList), this would not have been such a pain.

This was actually [a bug in flutter](https://github.com/flutter/flutter/pull/108710) - fixed now.

**Comment by baptistesx** [comment]
Hey @demchenkoalex,

I found your message here https://github.com/flyerhq/flutter_chat_ui/issues/287#issuecomment-1207181501 with the video in chat working.

i'm currently struggling with it.

Do i need to add a video player/controller?

I tried with this message (local uri): 
<img width="1276" alt="image" src="https://user-images.githubusercontent.com/9638281/209324535-28b306fa-4192-4394-a191-2a26656065c0.png">

and another with a remote uri (.mp4):
<img width="1334" alt="image" src="https://user-images.githubusercontent.com/9638281/209326374-d31c9114-7227-4323-9a33-84bb6efb2312.png">

Do you see something wrong?

Thanks in advance

**Reply by demchenkoalex** [reply]
Yes, you need a whole new class, VideoMessage, with controller and all of that. I did it similarly to the ImageMessage. It is not part of a package yet, that video I did locally just to confirm that the fix for the list is working. I did not release it, because I just ran the video, and to release a full feature we need controls, preview and all that stuff. I never include unfinished features in releases.

**Reply by baptistesx** [reply]
Of course, I totally agree about all the checks before releasing a new feature.
Did you push that draft video feature on a branch? Otherwise i'll try to implement it my self

**Reply by demchenkoalex** [reply]
Nope, I deleted all the code :D it was too "draft"
